src/core/src/op/unique.cpp

```diff
@@ -14,6 +15,80 @@ int64_t extract_axis(const std::shared_ptr<op::v0::Constant>& axis_constant) {
     const auto axis_vec = axis_constant->cast_vector<int64_t>();
     return axis_vec.at(0);
 }
+
+template <typename T, typename Index_t = int32_t, typename Counts_t = int32_t>
+ngraph::runtime::reference::UniqueElements<Index_t, Counts_t> call_unique(const Tensor& input,
+                                                                          std::unique_ptr<int64_t> axis,
+                                                                          const bool sorted) {
+    return ngraph::runtime::reference::find_unique_elements<T, Index_t, Counts_t>(input.data<T>(),
+                                                                                  input.get_shape(),
+                                                                                  std::move(axis),
+                                                                                  sorted);
+}
+
+std::tuple<Shape, Shape, Shape> calculate_static_output_shapes(const Tensor& input_data, const op::v10::Unique& op) {
+    using Index_t = int32_t;
+    using Counts_t = int32_t;
+
+    const auto maybe_extract_axis = [&op]() {
+        std::unique_ptr<int64_t> axis;
+        if (op.get_input_size() == 2 && ov::op::util::is_constant(op.input_value(1).get_node())) {
+            const auto axis_constant =
+                std::dynamic_pointer_cast<op::v0::Constant>(op.input_value(1).get_node_shared_ptr());
+            axis = std::unique_ptr<int64_t>(new int64_t{extract_axis(axis_constant)});
+        }
+        return axis;
+    };
+
+    ngraph::runtime::reference::UniqueElements<Index_t, Counts_t> unique_elements;
+    std::unique_ptr<int64_t> axis = maybe_extract_axis();
+
+    switch (op.get_input_element_type(0)) {
```

Comment:
looks like Android build is broken by this line https://dev.azure.com/openvinoci/dldt/_build/results?buildId=432415&view=logs&j=92ba3447-fcaa-530d-6438-b829fe1e7d1d&t=b0a91e75-8666-5718-e63e-df215ecb48f9&l=1404
The same for MacOS https://dev.azure.com/openvinoci/dldt/_build/results?buildId=432622&view=logs&j=a82d9e72-1de4-5bd6-c188-90296135479e&t=e36b0236-8854-50ba-7e69-658d6b092df0&l=1404

Please:
1. Fix these builds
2. Check whether we can add `-Wswitch` option to Ubuntu build (or find analogue flags for gcc) in order to align compilation behavior